src/unitxt/metrics.py

```diff
@@ -82,43 +83,98 @@ def _can_compute_confidence_intervals(self, num_predictions):
+         so they don't need to be recomputed after every bootstrap draw.
+
         score_names: List[str]
             Compute a confidence interval for each score_name from this list.
         instances:
             The instances for which the confidence intervals are computed.
+        aggregation_func:
+            A function with arguments instances, field_name; is applied on list of instances (which may include additional_inputs
+            field, as well as the prediction and references), and the field_name; default is simply to take the mean field_name from
+            instances after resampling, if aggregation_func=None.
         """
-        from statistics import mean
-
         result = {}
 
         if not self._can_compute_confidence_intervals(num_predictions=len(instances)):
             return result
+        # resample the indices of the instances, which contain the scores
+        identifiers = list(range(len(instances)))
 
-        score_names = (
-            self.ci_scores if self.ci_scores is not None else [self.main_score]
-        )
+        if score_names is None:
+            score_names = (
+                self.ci_scores if self.ci_scores is not None else [self.main_score]
+            )
+        if aggregation_func is None:
+
+            def aggregation_func(instances, field_name):
+                return mean([instance[field_name] for instance in instances])
 
         for score_name in score_names:
-            scores = [
-                instance["score"]["instance"][score_name] for instance in instances
-            ]
+            # need to redefine the statistic function within the loop because score_name is a loop variable, to avoid ruff errors
+            def statistic(arr, axis, score_name=score_name):
+                # arr is a 2d array where each row is a resampling, so we
+                # iterate over the rows and compute the metric on each resampling
+                def metric(instances):
+                    try:
+                        return aggregation_func(instances, score_name)
+                    except Exception as e:
+                        # this happens in edge cases, for example, when the sampling creates a
+                        # sample where all strings are empty and this fails bleu.
+                        logger.info(f"Warning in {self.__class__.__name__}", e)
+                        return np.nan
+
+                scores = numpy.apply_along_axis(
+                    lambda x: metric(
+                        [instances[ii] for ii in x],
+                    ),
+                    axis=axis,
+                    arr=arr,
+                )
+
+                return self.resample_from_non_nan(scores)
+
+            # apply bootstrap only on the relevant field
             ci = bootstrap(
-                (scores,),
-                statistic=mean,
+                (identifiers,),
```

Comment:
can you please explain why pass the `identifiers` and not the `instances` here?

Reply:
You're right, I found a work-around.  I had copied this part of the code from the GlobalMetric (which might not need it either).